### ISSUE TITLE
`org.ballerinalang.test.worker.WorkerFlushTest#errorBeforeFlush` fails intermittently

### ISSUE THREAD

**Issue by MaryamZi** [ballerina-platform/ballerina-lang#35670]
**Description:**
$title.

```
ballerina-new-parser-test-suite > jballerina-test > org.ballerinalang.test.worker.WorkerFlushTest > errorBeforeFlush FAILED
    org.testng.internal.thread.ThreadTimeoutException: Method org.ballerinalang.test.worker.WorkerFlushTest.errorBeforeFlush() didn't finish within the time-out 120000
        at java.base@11.0.11/jdk.internal.misc.Unsafe.park(Native Method)
        at java.base@11.0.11/java.util.concurrent.locks.LockSupport.park(LockSupport.java:194)
        at java.base@11.0.11/java.util.concurrent.locks.AbstractQueuedSynchronizer.parkAndCheckInterrupt(AbstractQueuedSynchronizer.java:885)
        at java.base@11.0.11/java.util.concurrent.locks.AbstractQueuedSynchronizer.acquireQueued(AbstractQueuedSynchronizer.java:917)
        at java.base@11.0.11/java.util.concurrent.locks.AbstractQueuedSynchronizer.acquire(AbstractQueuedSynchronizer.java:1240)
        at java.base@11.0.11/java.util.concurrent.locks.ReentrantLock.lock(ReentrantLock.java:267)
        at app//io.ballerina.runtime.internal.scheduling.WorkerDataChannel.acquireChannelLock(WorkerDataChannel.java:66)
        at app//io.ballerina.runtime.internal.scheduling.WorkerDataChannel.removeFlushWait(WorkerDataChannel.java:251)
        at app//io.ballerina.runtime.internal.scheduling.Strand.cleanUpFlush(Strand.java:241)
        at app//io.ballerina.runtime.internal.scheduling.Strand.handleFlush(Strand.java:208)
        at flush-workers.$lambda$_8(flush-workers.bal:107)
        at flush-workers.$lambda$_8$lambda7$(flush-workers.bal:100)
        at flush-workers$$Lambda$2550/0x000000010156a040.apply(Unknown Source)
        at app//io.ballerina.runtime.internal.scheduling.SchedulerItem.execute(Scheduler.java:610)
        at app//io.ballerina.runtime.internal.scheduling.Scheduler.run(Scheduler.java:316)
        at app//io.ballerina.runtime.internal.scheduling.Scheduler.runSafely(Scheduler.java:283)
        at app//io.ballerina.runtime.internal.scheduling.Scheduler.start(Scheduler.java:270)
        at app//org.ballerinalang.test.BRunUtil.invoke(BRunUtil.java:224)
        at app//org.ballerinalang.test.BRunUtil.invokeOnJBallerina(BRunUtil.java:150)
        at app//org.ballerinalang.test.BRunUtil.invoke(BRunUtil.java:87)
        at app//org.ballerinalang.test.BRunUtil.invoke(BRunUtil.java:289)
        at app//org.ballerinalang.test.worker.WorkerFlushTest.errorBeforeFlush(WorkerFlushTest.java:62)
```

**Comment by Nadeeshan96** [comment]
Seems this can be always reproduced by the following code which hangs giving the below output.
```bal
public function main() {
    error? innerResult = inner();
    error? innerOut = inner();
}

function inner() returns error? {
    worker w1 returns error? {
        int a = 10;
        error? sync = a ->> w3;
        return sync;
    }

    worker w3 returns error|string {
        int k = 5;
        if (k > 3) {
            map<string> reason = {k1: "error3"};
            error er3 = error(reason.get("k1"), message = "msg3");
            return er3;
        }
        int b = <- w1;
        return "done";
    }

    error? res = wait w1;
    return res;
}
```
```
Running executable

[2022-04-28 17:43:20,547] SEVERE {b7a.log.crash} - null 
java.lang.NullPointerException
        at io.ballerina.runtime.internal.scheduling.WorkerDataChannel.setReceiverPanic(WorkerDataChannel.java:295)
        at io.ballerina.runtime.internal.scheduling.Scheduler.notifyChannels(Scheduler.java:477)
        at io.ballerina.runtime.internal.scheduling.Scheduler.run(Scheduler.java:319)
        at io.ballerina.runtime.internal.scheduling.Scheduler.runSafely(Scheduler.java:283)
        at java.base/java.lang.Thread.run(Thread.java:829)
 
```

**Comment by Nadeeshan96** [comment]
Fixed with #36043.
